### PR TITLE
Add max swaps

### PR DESCRIPTION
**Pull request by peachbits**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203660431077319

### CODE REVIEW

File: src/components/scenes/CryptoExchangeScene.tsx [hunk 1]
```diff
@@ -212,37 +212,39 @@ export class CryptoExchangeComponent extends React.Component<Props, State> {
     return false
   }
 
-  getQuote = () => {
+  onNext = () => {
```
Comment (swansontec) [inline src/components/scenes/CryptoExchangeScene.tsx:204]: This should be called `handleNext`. There is a lint rule we eventually want to enable that requires this naming like `<X onFoo={handleWhatever} />` for all JSX callbacks.

Otherwise, this is a fantastic cleanup!

File: src/components/scenes/CryptoExchangeScene.tsx [hunk 2]
```diff
@@ -400,9 +415,7 @@ export class CryptoExchangeComponent extends React.Component<Props, State> {
             focusMe={this.focusFromWallet}
             onNext={this.onNext}
           >
-            {this.props.hasMaxSpend && (
-              <MiniButton alignSelf="center" label={s.strings.string_max_cap} marginRem={[1.2, 0, 0]} onPress={this.props.exchangeMax} />
-            )}
+            {this.props.hasMaxSpend && <MiniButton alignSelf="center" label={s.strings.string_max_cap} marginRem={[1.2, 0, 0]} onPress={this.onMax} />}
```
Comment (swansontec) [inline src/components/scenes/CryptoExchangeScene.tsx:392]: Optional: Since you are editing this line, it might be good to fix it to `hasMaxSpend ? <MiniButton ... > : null`. That way we are being explicit about rendering or not rendering, vs abusing the `&&` operator.

File: src/components/scenes/CryptoExchangeScene.tsx [hunk 3]
```diff
@@ -408,7 +382,7 @@ export class CryptoExchangeComponent extends React.Component<Props, State> {
             secondaryCurrencyInfo={fromSecondaryInfo}
             fiatPerCrypto={this.props.fromFiatToCrypto}
             overridePrimaryExchangeAmount={this.state.fromExchangeAmount}
-            forceUpdateGuiCounter={this.state.forceUpdateGuiCounter}
+            forceUpdateGuiCounter={0}
```
Comment (swansontec) [inline src/components/scenes/CryptoExchangeScene.tsx:385]: This prop is now unused. You can delete it from `CryptoExchangeFlipInputWrapper` and even `ExchangedFlipInput` itself.

File: src/components/scenes/CryptoExchangeScene.tsx [hunk 4]
```diff
@@ -179,25 +172,6 @@ export class CryptoExchangeComponent extends React.Component<Props, State> {
...
-        forceUpdateGuiCounter: props.forceUpdateGuiCounter
-      }
-    } else {
-      // Check which wallet we are currently editing.
-      // Only change the exchangeAmount of the opposite wallet to prevent feedback loops
-      if (state.whichWalletFocus === 'from') {
-        return { toExchangeAmount: props.toExchangeAmount }
-      } else {
-        return { fromExchangeAmount: props.fromExchangeAmount }
-      }
```
Comment (swansontec) [inline src/components/scenes/CryptoExchangeScene.tsx:197]: Deleting this `else` logic worries me. It doesn't seem related to the max send feature, but to general flip input instability.

File: src/constants/WalletAndCurrencyConstants.ts [hunk 5]
```diff
@@ -300,7 +300,6 @@ export const SPECIAL_CURRENCY_INFO: {
     initWalletName: s.strings.string_first_monero_wallet_name,
     chainCode: 'XMR',
     dummyPublicAddress: '46qxvuS78CNBoiiKmDjvjd5pMAZrTBbDNNHDoP52jKj9j5mk6m4R5nU6BDrWQURiWV9a2n5Sy8Qo4aJskKa92FX1GpZFiYA',
-    noMaxSpend: true,
```
Comment (swansontec) [inline src/constants/WalletAndCurrencyConstants.ts:303]: Assuming we merge the other Monero PR's, this commit message needs to change.

**Review by swansontec — CHANGES_REQUESTED** [review]

File: src/components/scenes/CryptoExchangeScene.tsx (hunk 4 above)
Comment (peachbits) [reply src/components/scenes/CryptoExchangeScene.tsx:197]: Good catch, reverted that part

File: src/components/scenes/CryptoExchangeScene.tsx [hunk 6]
```diff
@@ -212,6 +212,21 @@ export class CryptoExchangeComponent extends React.Component<Props, State> {
     return false
   }
 
+  onMax = () => {
```
Comment (swansontec) [inline src/components/scenes/CryptoExchangeScene.tsx:197]: `onMax` should be called `handleMax`.

**Review by swansontec — CHANGES_REQUESTED** [review]

**Review by swansontec — APPROVED** [review]
Thanks, this looks great now!
